configs/common/packages.yaml

```diff
@@ -9,290 +9,255 @@
     bufr:
-      version: ['12.0.1']
-      variants: +python
+      require: '@12.0.1 +python'
     cairo:
-      variants: +pic
+      require: '+pic'
     cdo:
-      version: ['2.2.0']
-      variants: ~openmp
+      require: '@2.2.0 ~openmp'
     cmake:
       version: ['3.27.9']
-      variants: +ownlibs
+      require: '+ownlibs'
     # Attention - when updating also check the various jcsda-emc-bundles env packages
     crtm:
-      version: ['2.4.0.1']
-      variants: +fix
+      require: '+fix'
     ecbuild:
-      version: ['3.7.2']
+      require: '@3.7.2'
     eccodes:
-      version: ['2.33.0']
-      variants: +png
+      require: '@2.33.0 +png'
     ecflow:
-      version: ['5.11.4']
-      variants: +ui
+      require: '@5.11.4 +ui'
     eckit:
-      version: ['1.24.5']
-      variants: linalg=eigen,lapack compression=lz4,bzip2
+      require: '@1.24.5 linalg=eigen,lapack compression=lz4,bzip2'
     ecmwf-atlas:
-      version: ['0.36.0']
-      variants: +fckit +trans +tesselation +fftw
+      require: '@0.36.0 +fckit +trans +tesselation +fftw'
     ectrans:
-      version: ['1.2.0']
-      variants: ~mkl +fftw
+      require: '@1.2.0 ~mkl +fftw'
     eigen:
-      version: ['3.4.0']
+      require: '@3.4.0'
     # Attention - when updating the version also check the common modules.yaml
     # config and update the projections for lmod/tcl.
     # Also, check the acorn and derecho site configs which have esmf modifications.
     esmf:
       require:
-        - any_of: ['@=8.6.1 ~xerces ~pnetcdf snapshot=none +shared +external-parallelio fflags="-fp-model precise" cxxflags="-fp-model precise"']
+        - '@=8.6.1 ~xerces ~pnetcdf snapshot=none +shared +external-parallelio'
+        - any_of: ['fflags="-fp-model precise" cxxflags="-fp-model precise"']
           when: "%intel"
           message: "Extra ESMF compile options for Intel"
-        - any_of: ['@=8.6.1 ~xerces ~pnetcdf snapshot=none +shared +external-parallelio']
+        - any_of: ['']
           when: "%gcc"
           message: "Extra ESMF compile options for GCC"
-        - any_of: ['@=8.6.1 ~xerces ~pnetcdf snapshot=none +shared +external-parallelio']
+        - any_of: ['']
           when: "%apple-clang"
           message: "Extra ESMF compile options for GCC"
     fckit:
-      version: ['0.11.0']
-      variants: +eckit
+      require: '@0.11.0 +eckit'
     fftw:
-      version: ['3.3.10']
+      require: '@3.3.10'
     fiat:
-      version: ['1.2.0']
+      require: '@1.2.0'
+    flex:
+      # Pin version to avoid duplicates
+      require: '@2.6.4'
     fms:
-      version: ['2023.04']
-      variants: precision=32,64 +quad_precision +gfs_phys +openmp +pic constants=GFS build_type=Release +deprecated_io
+      variants: 'precision=32,64 +quad_precision +gfs_phys +openmp +pic constants=GFS build_type=Release +deprecated_io'
```

Comment:
Does this work with the vastly different versions of FMS, where the older ones  don't even know about some of these variants?